engine/classes/Elgg/Application.php

```diff
@@ -0,0 +1,293 @@
+			'notification.php',
+			'objects.php',
+			'output.php',
+			'pagehandler.php',
+			'pageowner.php',
+			'pam.php',
+			'plugins.php',
+			'private_settings.php',
+			'relationships.php',
+			'river.php',
+			'sessions.php',
+			'sites.php',
+			'statistics.php',
+			'system_log.php',
+			'tags.php',
+			'user_settings.php',
+			'users.php',
+			'upgrade.php',
+			'views.php',
+			'widgets.php',
+
+			// backward compatibility
+			'deprecated-1.7.php',
+			'deprecated-1.8.php',
+			'deprecated-1.9.php',
+			'deprecated-1.10.php',
+		);
+
+		// isolate global scope
+		call_user_func(function () use ($lib_dir, $lib_files) {
+
+			$setups = array();
+
+			// include library files, capturing setup functions
+			foreach ($lib_files as $file) {
+				$setup = (require_once "$lib_dir/$file");
+
+				if ($setup instanceof \Closure) {
+					$setups[$file] = $setup;
+				}
+			}
+
+			$events = _elgg_services()->events;
+			$hooks = _elgg_services()->hooks;
+
+			// run setups
+			foreach ($setups as $func) {
+				$func($events, $hooks);
+			}
+		});
+	}
+
+	/**
+	 * Bootstraps the Elgg engine.
+	 *
+	 * This method loads the full Elgg engine, checks the installation
+	 * state, and triggers a series of events to finish booting Elgg:
+	 * 	- {@elgg_event boot system}
+	 * 	- {@elgg_event init system}
+	 * 	- {@elgg_event ready system}
+	 *
+	 * If Elgg is fully uninstalled, the browser will be redirected to an
+	 * installation page.
+	 *
+	 * @see install.php
+	 * @package Elgg.Core
+	 * @subpackage Core
+	 */
+	function bootCore() {
+		global $CONFIG;
+
+		$this->loadConfig();
+
+		// This will be overridden by the DB value but may be needed before the upgrade script can be run.
+		$CONFIG->default_limit = 10;
+
+		$this->loadCore();
+
+		// Connect to database, load language files, load configuration, init session
+		// Plugins can't use this event because they haven't been loaded yet.
+		elgg_trigger_event('boot', 'system');
+
+		// Load the plugins that are active
+		_elgg_load_plugins();
+
+		// @todo move loading plugins into a single boot function that replaces 'boot', 'system' event
+		// and then move this code in there.
+		// This validates the view type - first opportunity to do it is after plugins load.
+		$viewtype = elgg_get_viewtype();
+		if (!elgg_is_registered_viewtype($viewtype)) {
+			elgg_set_viewtype('default');
+		}
+
+		// @todo deprecate as plugins can use 'init', 'system' event
+		elgg_trigger_event('plugins_boot', 'system');
+
+		// Complete the boot process for both engine and plugins
+		elgg_trigger_event('init', 'system');
+
+		$CONFIG->boot_complete = true;
+
+		// System loaded and ready
+		elgg_trigger_event('ready', 'system');
+
+	}
+
+	/**
+	 * Rewrite rules for PHP cli webserver used for testing. Do not use on production sites
+	 * as normal web server replacement.
+	 *
+	 * You need to explicitly point to index.php in order for router to work properly:
+	 *
+	 * <code>php -S localhost:8888 index.php</code>
```

Comment:
Is there an easy way to set the web root when using this? I assume currently the external resource links are broken?